x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/conditional_tooltip.tsx

```diff
@@ -27,113 +27,83 @@ import { createFormatterForMetric } from '../../../metrics_explorer/components/h
 
 export interface Props {
   currentTime: number;
-  hidden: boolean;
   node: InfraWaffleMapNode;
-  options: InfraWaffleMapOptions;
-  formatter: (val: number) => string;
-  children: React.ReactElement;
   nodeType: InventoryItemType;
   theme: EuiTheme | undefined;
 }
-
-export const ConditionalToolTip = withTheme(
-  ({ theme, hidden, node, children, nodeType, currentTime }: Props) => {
-    const { sourceId } = useSourceContext();
-    const [timer, setTimer] = useState<ReturnType<typeof setTimeout> | null>(null);
-    const model = findInventoryModel(nodeType);
-    const { customMetrics } = useWaffleOptionsContext();
-    const requestMetrics = model.tooltipMetrics
-      .map((type) => ({ type }))
-      .concat(customMetrics) as Array<
-      | {
-          type: SnapshotMetricType;
-        }
-      | SnapshotCustomMetricInput
-    >;
-    const query = JSON.stringify({
-      bool: {
-        filter: {
-          match_phrase: { [model.fields.id]: node.id },
-        },
+export const ConditionalToolTip = withTheme(({ theme, node, nodeType, currentTime }: Props) => {
+  const { sourceId } = useSourceContext();
+  const model = findInventoryModel(nodeType);
+  const { customMetrics } = useWaffleOptionsContext();
+  const requestMetrics = model.tooltipMetrics
+    .map((type) => ({ type }))
+    .concat(customMetrics) as Array<
+    | {
+        type: SnapshotMetricType;
+      }
+    | SnapshotCustomMetricInput
+  >;
+  const query = JSON.stringify({
+    bool: {
+      filter: {
+        match_phrase: { [model.fields.id]: node.id },
       },
-    });
-    const { nodes, reload } = useSnapshot(
-      query,
-      requestMetrics,
-      [],
-      nodeType,
-      sourceId,
-      currentTime,
-      '',
-      '',
-      false // Doesn't send request until reload() is called
-    );
-
-    const handleDataLoad = useCallback(() => {
-      const id = setTimeout(reload, 200);
-      setTimer(id);
-    }, [reload]);
+    },
+  });
+  const { nodes, reload } = useSnapshot(
+    query,
+    requestMetrics,
+    [],
+    nodeType,
+    sourceId,
+    currentTime,
+    '',
+    '',
+    false // Doesn't send request until reload() is called
+  );
 
-    const cancelDataLoad = useCallback(() => {
-      return (timer && clearTimeout(timer)) || void 0;
-    }, [timer]);
+  useEffect(() => {
+    reload();
+  }, [reload]);
```

Comment:
I think you don't need this anymore, setting the snapshot request to be called immediately should be enough, or I'm missing something?

Reply:
updated! @estermv